contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+      }
+    }
+    return escrowScheduleDetails;
+  }
+
+  /**
+   * @notice Returns all governing settings for an escrow schedule.
+   */
+  function getEscrowSchedule(address account, uint256 escrowScheduleId)
+    public
+    view
+    returns (EscrowScheduleDetail memory)
+  {
+    EscrowSchedule storage escrowAcount = _addressToEscrowSchedules[account][
+      escrowScheduleId
+    ];
+    return
+      EscrowScheduleDetail(
+        escrowAcount.currentAmount,
+        account,
+        escrowAcount.startTime,
+        escrowAcount.endTime,
+        escrowAcount.claimedAmount,
+        escrowAcount.totalQuantityRevoked,
+        escrowAcount.exists
+      );
+  }
+
+  /**
+   * @notice Released balance less any claimed amount at current block timestamp.
+   */
+  function claimableBalanceOf(address account) public view returns (uint256) {
+    return _claimableBalanceOf(account, block.timestamp); // solhint-disable-line not-rely-on-time, this is time-dependent
+  }
+
+  /**
+   * @notice Overridden standard ERC777.burn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function burn(uint256, bytes memory) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Overridden standard ERC777.operatorBurn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function operatorBurn(
+    address,
+    uint256,
+    bytes memory,
+    bytes memory
+  ) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Wraps minting of wrapper token and escrow schedule setup.
+   *
+   * @dev If `startTime` is zero no escrow schedule is set up.
+   * Satisfies situations where funding of the account happens over time.
+   *
+   * @param amount uint256 Quantity of `_bridgedPolygonNori` to deposit
+   * @param userData bytes extra information provided by the user (if any)
+   * @param operatorData bytes extra information provided by the operator (if any)
+   */
+  function _depositFor(
+    uint256 removalId,
+    uint256 amount,
+    bytes calldata userData,
+    bytes calldata operatorData
+  ) internal returns (bool) {
+    address recipient = removalId.supplierAddress();
+    uint256 escrowScheduleId = _removal.getEscrowScheduleIdForRemoval(
+      removalId
+    );
+    // TODO error handling/ what to do if this value doesn't exist??
+    // should we calculate the unix startTime for the removal's vintage? (or look it up)
+    if (!_addressToEscrowSchedules[recipient][escrowScheduleId].exists) {
+      _createEscrowSchedule(recipient, escrowScheduleId);
+    }
+    super._mint(recipient, amount, userData, operatorData);
+    EscrowSchedule storage escrowSchedule = _addressToEscrowSchedules[
+      recipient
+    ][escrowScheduleId];
+    escrowSchedule.currentAmount += amount;
```

Comment:
maybe we should override the `_mint` function for this logic

Reply:
do you mean basically put the contents of `_depositFor` into an overridden `_mint` ?